apps/server/src/shared/domain/entity/course.entity.ts

```diff
@@ -68,12 +68,9 @@ export class Course extends BaseEntityWithTimestamps {
 		return this.students.length;
 	}
 
-	isSubstitutionTeacher(userId: EntityId): boolean {
-		const teacherIds = this.teachers.getIdentifiers('id');
+	getSubstitutionTeacherIds(): EntityId[] {
 		const substitutionIds = this.substitutionTeachers.getIdentifiers('id');
-
-		const isSubstitutionTeacher = !teacherIds.includes(userId) && substitutionIds.includes(userId);
-
-		return isSubstitutionTeacher;
+		// The result of getIdentifiers is a primary key type where we have no represent for it ((string | ObjectId) & IPrimaryKeyValue)[]
+		return substitutionIds.filter((id) => id).map((id) => id.toString());
```

Comment:
if we have to map the result anyway, why not write:
```typescript
const substitutionIds = this.substitutionTeachers.map((o) => o.id);
return substitutionIds;
```
the assignment also helps us when debugging.